entities/JsonHandlers/parsers/EntityJsonParser.cpp

```diff
@@ -51,6 +51,8 @@ std::unique_ptr<Way> EntityJsonParser::parseWay(rapidjson::Value &doc, const GIS
     std::unique_ptr<Way> way = std::make_unique<Way>(id, name, description, categoryTags, std::move(geometry), from, to,
                                                      direction, speedLimit, tollRoad,
                                                      restricted);
+    ((Junction *)fromEntity)->addWay(id);
+    ((Junction *)toEntity)->addWay(id);
```

Comment:
We only want to add it to the end junction if it is bidirectional

Reply:
fixed